### PR TITLE
AMLS-4384: PPoB Non UK allows user to select United Kingdom.

### PR DESCRIPTION
**Pull request by drewmiley**
PPoB Non UK allows user to select United Kingdom. This PR is to fix this issue alongside adding new error messages.

## How Has This Been Tested?
Unit testing
Manual Testing

## Checklist

- [ ] Breaking change (fix or feature that would cause existing functionality to change).
- [x] Build passing.
- [x] Unit tests have full coverage.
- [ ] Unit test approach and implementation has been reviewed with QA (testers).
- [ ] Requires acceptance test run.
- [x] Appropriate labels added.
- [x] RELEASE NOTES ADDED.

### CODE REVIEW

File: test/models/responsiblepeople/ResponsiblePersonAddressHistorySpec.scala [hunk 1]
```diff
@@ -25,12 +25,12 @@ import play.api.libs.json.Json
...
   val DefaultAdditionalExtraAddress = ResponsiblePersonAddress(PersonAddressUK("Line 1", "Line 2", None, None, "NE1234"), Some(OneToThreeYears))
 
-  val NewCurrentAddress = ResponsiblePersonCurrentAddress(PersonAddressNonUK("Line 1", "Line 2", None, None, Country("Spain", "ES")), Some(ZeroToFiveMonths))
-  val NewAdditionalAddress = ResponsiblePersonAddress(PersonAddressNonUK("Line 1", "Line 2", None, None, Country("France", "FR")), Some(ZeroToFiveMonths))
-  val NewAdditionalExtraAddress = ResponsiblePersonAddress(PersonAddressNonUK("Line 1", "Line 2", None, None, Country("UK", "UK")), Some(SixToElevenMonths))
+  val NewCurrentAddress = ResponsiblePersonCurrentAddress(PersonAddressNonUK("Line 1", "Line 2", None, None, NonUKCountry("Spain", "ES")), Some(ZeroToFiveMonths))
+  val NewAdditionalAddress = ResponsiblePersonAddress(PersonAddressNonUK("Line 1", "Line 2", None, None, NonUKCountry("France", "FR")), Some(ZeroToFiveMonths))
+  val NewAdditionalExtraAddress = ResponsiblePersonAddress(PersonAddressNonUK("Line 1", "Line 2", None, None, NonUKCountry("UK", "UK")), Some(SixToElevenMonths))
```
Comment (drewmiley) [inline test/models/responsiblepeople/ResponsiblePersonAddressHistorySpec.scala]: That's not ok

File: test/models/responsiblepeople/PersonAddressSpec.scala [hunk 2]
```diff
@@ -16,7 +16,7 @@
 
 package models.responsiblepeople
 
-import models.Country
+import models.{Country, NonUKCountry}
```
Comment (drewmiley) [inline test/models/responsiblepeople/PersonAddressSpec.scala]: Is this import Country required

File: test/models/businessdetails/RegisteredOfficeSpec.scala [hunk 3]
```diff
@@ -285,7 +301,7 @@ class RegisteredOfficeSpec extends PlaySpec with MockitoSugar {
     "convert Business Customer Address to RegisteredOfficeNonUK" in {
       val address = Address("addr1", "addr2", Some("line3"), Some("line4"), None, Country("United Kingdom", "GB"))
 
-      RegisteredOffice.convert(address) must be(RegisteredOfficeNonUK("addr1", "addr2", Some("line3"), Some("line4"), Country("United Kingdom", "GB")))
+      RegisteredOffice.convert(address) must be(RegisteredOfficeNonUK("addr1", "addr2", Some("line3"), Some("line4"), NonUKCountry("United Kingdom", "GB")))
```
Comment (drewmiley) [inline test/models/businessdetails/RegisteredOfficeSpec.scala]: That's not ok

File: test/audit/AddressCreatedEventSpec.scala [hunk 4]
```diff
@@ -90,7 +90,7 @@ class AddressCreatedEventSpec extends AmlsSpec {
       }
 
       "given the address of a registered office outside the UK" in {
-        val address = RegisteredOfficeNonUK("RO Line 1", "RO Line 2", "RO Line 3".some, None, Country("Scotland", "SCO"))
+        val address = RegisteredOfficeNonUK("RO Line 1", "RO Line 2", "RO Line 3".some, None, NonUKCountry("Scotland", "SCO"))
```
Comment (drewmiley) [inline test/audit/AddressCreatedEventSpec.scala]: Change

File: app/models/responsiblepeople/PersonAddress.scala [hunk 5]
```diff
@@ -16,7 +16,7 @@
 
 package models.responsiblepeople
 
-import models.Country
+import models.{Country, NonUKCountry}
```
Comment (drewmiley) [inline app/models/responsiblepeople/PersonAddress.scala]: Import?

File: app/models/package.scala [hunk 6]
```diff
@@ -449,4 +449,259 @@ package object models {
     Country("Zambia", "ZM"),
     Country("Zimbabwe", "ZW")
   )
+
+  val nonUkcountries = Seq(
```
Comment (drewmiley) [inline app/models/package.scala]: Remove duplication

File: app/models/businessdetails/RegisteredOffice.scala [hunk 7]
```diff
@@ -175,7 +174,7 @@ object RegisteredOffice {
         address.line_2,
         address.line_3,
         address.line_4,
-        address.country)
+        new NonUKCountry(address.country.name, address.country.code))
```
Comment (drewmiley) [inline app/models/businessdetails/RegisteredOffice.scala]: Investigate

File: app/models/NonUKCountry.scala [hunk 8]
```diff
@@ -0,0 +1,70 @@
+/*
```
Comment (drewmiley) [inline app/models/NonUKCountry.scala]: Duplication possible to avoid

File: app/models/responsiblepeople/PersonAddress.scala (hunk 5 above)
Comment (drewmiley) [reply app/models/responsiblepeople/PersonAddress.scala]: need

File: app/controllers/responsiblepeople/CountryOfBirthController.scala [hunk 9]
```diff
@@ -37,7 +36,7 @@ class CountryOfBirthController @Inject()(val authConnector: AuthConnector,
 
   private def getCountryOfBirth(countryOfBirth: Country): CountryOfBirth = {
      if(countryOfBirth.code != "GB") {
-        CountryOfBirth(false, Some(countryOfBirth))
+        CountryOfBirth(false, Some(Country(countryOfBirth.name, countryOfBirth.code)))
```
Comment (matthewharris1) [inline app/controllers/responsiblepeople/CountryOfBirthController.scala]: revert

**Comment by drewmiley** [comment]
I'll retest this in the morning- spotted a code compile issue

**Comment by drewmiley** [comment]
I think a bracket is missing from the end of this line possibly? https://github.com/hmrc/amls-frontend/pull/1067/commits/0ed0185451b7a8accdc68f4052873f318a0468ea#diff-55ba8ca956a262419c30e0c233e7dcf4

**Pull request merged**